crystal-map-processor/src/main/java/com/schwarz/crystalprocessor/generation/model/SchemaGeneration.kt

```diff
@@ -0,0 +1,176 @@
+package com.schwarz.crystalprocessor.generation.model
+
+import com.schwarz.crystalapi.schema.*
+import com.schwarz.crystalprocessor.model.entity.SchemaClassHolder
+import com.schwarz.crystalprocessor.model.field.CblBaseFieldHolder
+import com.schwarz.crystalprocessor.model.field.CblFieldHolder
+import com.schwarz.crystalprocessor.util.ConversionUtil
+import com.schwarz.crystalprocessor.util.TypeUtil
+import com.squareup.kotlinpoet.*
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+
+/**
+ * This class is responsible for generating the Schema classes.
+ *
+ * To generate a SchemaClass, add the following annotation to your Class:
+ * ```
+ * @SchemaClass
+ * ```
+ * All the fields will then be generated into a new file.
+ */
+class SchemaGeneration {
+    private val pathAttributeName = "path"
+    fun generateModel(holder: SchemaClassHolder, schemaClassPaths: List<String>): FileSpec {
+        val packageName = holder.sourcePackage
+        val schemaClassName = holder.entitySimpleName
+
+        val schemaClass: TypeSpec.Builder = buildSchemaClass(schemaClassName)
+
+        buildAndAddFieldProperties(holder, schemaClass, schemaClassPaths)
+
+        return FileSpec.builder(packageName, schemaClassName).addType(schemaClass.build()).build()
+    }
+
+    private fun buildSchemaClass(className: String): TypeSpec.Builder {
+        val pathParameter = ParameterSpec.builder(pathAttributeName, String::class).defaultValue("%S", "").build()
+
+        return TypeSpec.classBuilder(className)
+            .addModifiers(KModifier.OPEN)
```

Comment:
What ClassName do we expect here as a final result? Code sounds like the same as the resulting entity/wrapper?

Reply:
When, for example, the entity named "Article" has the @SchemaClass annotation, the final class name would be "ArticleSchema".